### PR TITLE
Add ability to generate particular types as Java interfaces

### PR DESCRIPTION
**Pull request by donbeave**
Implementation for this issue: https://github.com/kobylynskyi/graphql-java-codegen/issues/606

### CODE REVIEW

**Comment by donbeave** [comment]
@kobylynskyi please check, only one task is failed is sonar task: https://github.com/kobylynskyi/graphql-java-codegen/runs/2186852139

But it looks like it's CI configuration issue, not something related to my changes in this PR.

**Comment by kobylynskyi** [comment]
Yes, Sonar fails for all PRs from forked repos. 🥴

**Review by kobylynskyi — CHANGES_REQUESTED** [review]
@donbeave good job! Just one minor comment related to the codegen-options.md.
Once it is done - I am going to merge this to develop branch and release as part of 4.1.6.
Thanks again for the contribution! Much appreciate it.

File: docs/codegen-options.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ See [DirectiveAnnotationsMapping](#option-directiveannotationsmapping)* |
...
 | `generatedLanguage`                                   | Enum                                                                  | GeneratedLanguage.JAVA                                 | Choose which language you want to generate, Java,Scala,Kotlin were supported. Note that due to language features, there are slight differences in default values between languages.|
 | `generateModelOpenClasses`                            | Boolean                                                               | false                                                  | The class type of the generated model. If true, generate normal classes, else generate data classes. It only support in kotlin(```data class```) and scala(```case class```). Maybe we will consider to support Java ```record``` in the future.|
+| `typeInterfaces`                                      | Set(String)                                                           | Empty                                                  | Types that must generated as interfaces should be defined here in format: `TypeName` or `@directive`. E.g.: `User`, `@customInterface`. |
```
[the remark this answers is not in the document]
Comment (jxnu-liguobin) [reply docs/codegen-options.md:57]: There may be ambiguity when use `@customInterface`, consider using `@asInterface` ?

[the remark this answers is not in the document]
Comment (donbeave) [reply docs/codegen-options.md]: @kobylynskyi correct, it's just a typo
@jxnu-liguobin thank you! nice catch!

**Review by kobylynskyi — APPROVED** [review]

**Pull request merged**